lib/statsd/instrument/compiled_metric.rb

```diff
@@ -97,16 +98,47 @@ def define_metric_method(tags)
 
         private
 
+        def generate_block_handler
+          <<~RUBY
+            # For all timing metrics, we have to use the sampling logic.
+            # Not doing so would impact performance and CPU usage.
+            # See Datadog's documentation for more details: https://github.com/DataDog/datadog-go/blob/20af2dbfabbbe6bd0347780cd57ed931f903f223/statsd/aggregator.go#L281-L283
+            sample_rate ||= @default_sample_rate
```

Comment:
nit: Later on, this wont be read by any human, will it? Does it make sense to comment the meta programming code?